be/src/util/rle_encoding.h

```diff
@@ -886,4 +891,68 @@ inline int32_t RleBatchDecoder<T>::GetBatch(T* values, int32_t batch_num) {
     return num_consumed;
 }
 
+template <typename T>
+static inline bool IndexInRange(T idx, int32_t dictionary_length) {
+    return idx >= 0 && idx < dictionary_length;
+}
+
+template <typename T>
+static inline bool IndicesInRange(const T* values, int32_t length, int32_t dictionary_length) {
+    T min_index = std::numeric_limits<T>::max();
+    T max_index = std::numeric_limits<T>::min();
+    for (int x = 0; x < length; x++) {
+        min_index = std::min(values[x], min_index);
+        max_index = std::max(values[x], max_index);
+    }
+
+    return IndexInRange(min_index, dictionary_length) && IndexInRange(max_index, dictionary_length);
+}
+
+template <typename T>
+template <typename TV>
+inline int RleBatchDecoder<T>::GetBatchWithDict(const TV* dictionary, int32_t dictionary_length, TV* values,
```

Comment:
template<typename T, typename TV>

Reply:
`typename TV` is a template variable of member method `GetBatchWithDict`, it's not legal to put them together as per [member_template](https://en.cppreference.com/w/cpp/language/member_template)